bridges/snowbridge/pallets/system-frontend/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+		Send,
+		/// Withdraw fee asset failure
+		FundsUnavailable,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Call `create_agent` to instantiate a new agent contract representing `origin`.
+		/// - `origin`: Can be any sovereign `Location`
+		/// - `fee`: Fee in Ether paying for the execution cost on Ethreum
+		#[pallet::call_index(1)]
+		#[pallet::weight(T::WeightInfo::create_agent())]
+		pub fn create_agent(origin: OriginFor<T>, fee: u128) -> DispatchResult {
+			let origin_location = T::CreateAgentOrigin::ensure_origin(origin)?;
+
+			Self::burn_fees(origin_location.clone(), fee)?;
+
+			let call = SnowbridgeControl::Control(ControlCall::CreateAgent {
+				location: Box::new(VersionedLocation::from(origin_location.clone())),
+				fee,
+			});
+
+			let xcm: Xcm<()> = vec![
+				ReceiveTeleportedAsset(T::DeliveryFee::get().into()),
+				PayFees { asset: T::DeliveryFee::get() },
+				Transact {
+					origin_kind: OriginKind::Xcm,
+					call: call.encode().into(),
+					fallback_max_weight: None,
+				},
+			]
+			.into();
+
+			Self::send(xcm)?;
+
+			Self::deposit_event(Event::<T>::CreateAgent { location: origin_location.clone() });
+			Ok(())
+		}
+
+		/// Registers a Polkadot-native token as a wrapped ERC20 token on Ethereum.
+		/// - `origin`: Must be `Location` from a sibling parachain
+		/// - `asset_id`: Location of the asset (should starts from the dispatch origin)
+		/// - `metadata`: Metadata to include in the instantiated ERC20 contract on Ethereum
+		/// - `fee`: Fee in Ether paying for the execution cost on Ethreum
+		#[pallet::call_index(2)]
+		#[pallet::weight(T::WeightInfo::register_token())]
+		pub fn register_token(
+			origin: OriginFor<T>,
+			asset_id: Box<VersionedLocation>,
+			metadata: AssetMetadata,
+			fee: u128,
+		) -> DispatchResult {
+			let origin_location = T::RegisterTokenOrigin::ensure_origin(origin)?;
+
+			let asset_location: Location =
+				(*asset_id).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
+
+			let mut checked = false;
+			if asset_location.eq(&origin_location) || asset_location.starts_with(&origin_location) {
+				checked = true
+			}
+			ensure!(checked, <Error<T>>::OwnerCheck);
+
+			Self::burn_fees(origin_location.clone(), fee)?;
+
+			let call = SnowbridgeControl::Control(ControlCall::RegisterToken {
+				asset_id: Box::new(VersionedLocation::from(asset_location.clone())),
+				metadata,
+				fee,
+			});
+
+			let xcm: Xcm<()> = vec![
+				ReceiveTeleportedAsset(T::DeliveryFee::get().into()),
+				PayFees { asset: T::DeliveryFee::get() },
+				Transact {
+					origin_kind: OriginKind::Xcm,
+					call: call.encode().into(),
+					fallback_max_weight: None,
+				},
+			]
+			.into();
+
+			Self::send(xcm)?;
+
+			Self::deposit_event(Event::<T>::RegisterToken { location: asset_location });
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub fn send(xcm: Xcm<()>) -> DispatchResult {
+			let bridgehub = Location::new(1, [Parachain(1002)]);
```

Comment:
@yrong Rather pass this location via config please

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/e03618b8b4ec4b3b7851099b0bd5f5a1b7c19130